### DISCUSSION TITLE
Add 'styles' to the CKE5 version of RTE

### DISCUSSION THREAD

**Discussion by pyrobob83** [ExpressionEngine/ExpressionEngine#3136] [Feature Requests]
CKE5 finally has 'Styles' per https://ckeditor.com/docs/ckeditor5/latest/features/style.html.  The docs state "The style feature is not available in any of the predefined builds", so I assume we'd have to manually implement it to make it available, but I'd like to request we do this.

For me the primary objective is to replicate what's available in Wygwam - specifically the ability to add styling to elements via classes with options very specific to particular HTML elements.  Using this JS configuration in Wygwam:
```CKEDITOR.stylesSet.add( 'my_styles',
[
    // Block Styles
    { name: 'Lead Text', element: 'p', attributes: { 'class': 'lead' } },
    { name: 'Heading 2' , element: 'h2' },
    { name: 'Heading 3' , element: 'h3' },
    //Inline Styles
	{ name: 'Btn' , element: 'a', attributes: { 'class': 'btn' } },
	{ name: 'Btn Small' , element: 'a', attributes: { 'class': 'btn btn--small' } },
]);
```
You only get the Btn and Btn Small options if you're highlighting or your cursor is in an <a> tag.

I have no idea what it'd take or what the updated configuration would need to look like, but I hope it's possible!  Thanks!

**Comment by intoeetive** [comment]
This is getting added in https://github.com/ExpressionEngine/ExpressionEngine/pull/3586
(you'd need to use advanced configuration JSON to implement this)
